### PR TITLE
KAFKA-8454: Add Java AdminClient Interface (KIP-476)

### PR DESCRIPTION
**Pull request by big-andy-coates**
[KIP-476](https://cwiki.apache.org/confluence/display/KAFKA/KIP-476%3A+Add+Java+AdminClient+Interface)

Adds a `Admin` interface.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by big-andy-coates** [comment]
ping @cmccabe, @mjsax, @rajinisivaram  for a review please

I've added the `Admin` interface, removed most of the `AdminClient` impl, and updates references in the docs to `AdminClient` where I think it's appropriate.

File: clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java [hunk 1]
```diff
@@ -17,926 +17,36 @@
...
 
     /**
-     * Create a new AdminClient with the given configuration.
+     * Create a new Admin with the given configuration.
      *
      * @param props The configuration.
      * @return The new KafkaAdminClient.
      */
-    public static AdminClient create(Properties props) {
+    public static Admin create(Properties props) {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java]: This is a breaking change, right? Same for the other `create` method in this class.

Comment (big-andy-coates) [reply clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java]: Yep, sorry, my bad. Should have spotted this.

Comment (big-andy-coates) [reply clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java]: Fixed.

File: clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java [hunk 2]
```diff
@@ -17,38 +17,20 @@
...
+ * The base class for in-built admin clients.
  *
- * The minimum broker version required is 0.10.0.0. Methods with stricter requirements will specify the minimum broker
- * version required.
+ * Client code should use the newer {@link Admin} interface in preference to this class.
  *
- * This client was introduced in 0.11.0.0 and the API is still evolving. We will try to evolve the API in a compatible
- * manner, but we reserve the right to make breaking changes in minor releases, if necessary. We will update the
- * {@code InterfaceStability} annotation and this notice once the API is considered stable.
+ * This class may be removed in a later release, but has not be marked as deprecated to avoid unnecessary noise.
```
Comment (hachikuji) [inline clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java:28]: This was probably discussed in the KIP, but obviously the downside is that users won't get any warning or hint that they should transition. But I guess we don't get a substantial benefit from removing `AdminClient`, so maybe we'll just never do it.

File: checkstyle/suppressions.xml [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
     <!-- Clients -->
     <suppress checks="ClassFanOutComplexity"
-              files="(Fetcher|Sender|SenderTest|ConsumerCoordinator|KafkaConsumer|KafkaProducer|Utils|TransactionManagerTest|KafkaAdminClient|NetworkClient|AdminClient).java"/>
+              files="(Fetcher|Sender|SenderTest|ConsumerCoordinator|KafkaConsumer|KafkaProducer|Utils|TransactionManagerTest|KafkaAdminClient|NetworkClient|Admin|Admin).java"/>
```
Comment (hachikuji) [inline checkstyle/suppressions.xml]: Why is `Admin` listed twice here and below?

**Review by hachikuji — COMMENTED** [review]
Left a couple minor comments, but looks good overall.

File: clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java (hunk 2 above)
Comment (big-andy-coates) [reply clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java:28]: I agree, but this was covered / discussed in detail on the voting thread and this was the outcome.

File: checkstyle/suppressions.xml (hunk 3 above)
Comment (big-andy-coates) [reply checkstyle/suppressions.xml]: Not intentional.  Will fix.

**Comment by hachikuji** [comment]
retest this please

**Review by hachikuji — APPROVED** [review]
LGTM

**Comment by hachikuji** [comment]
retest this please

**Comment by big-andy-coates** [comment]
retest this please

**Comment by big-andy-coates** [comment]
retest this please

**Comment by big-andy-coates** [comment]
retest this please

**Pull request merged**